packages/plugin-patch/sources/patchUtils.ts

```diff
@@ -172,3 +172,18 @@ export async function extractPackageToDisk(locator: Locator, {cache, project}: {
   xfs.detachTemp(temp);
   return temp;
 }
+
+export async function diffFolders(folderA: PortablePath, folderB: PortablePath) {
+  const folderAN = npath.fromPortablePath(folderA).replace(/\\/g, `/`);
+  const folderBN = npath.fromPortablePath(folderB).replace(/\\/g, `/`);
+
+  const {stdout} = await execUtils.execvp(`git`, [`diff`, `--ignore-cr-at-eol`, `--full-index`, `--no-index`, folderAN, folderBN], {
+    cwd:npath.toPortablePath(process.cwd()),
+  });
+
+  return stdout
+    .replace(new RegExp(`(a|b)${miscUtils.escapeRegExp(`/${folderAN.substr(folderAN.startsWith(`/`) ? 1 : 0)}/`)}`, `g`), `$1/`)
+    .replace(new RegExp(`(a|b)${miscUtils.escapeRegExp(`/${folderBN.substr(folderBN.startsWith(`/`) ? 1 : 0)}/`)}`, `g`), `$1/`)
+    .replace(new RegExp(miscUtils.escapeRegExp(`${folderAN}/`), `g`), ``)
+    .replace(new RegExp(miscUtils.escapeRegExp(`${folderBN}/`), `g`), ``);
```

Comment:
Why are those two lines needed, since we cover `a/` and `b/` right above?

Reply:
They are needed because the paths for a `rename` change does not start with neither `a/` nor `b/`